CMakeLists.txt

```diff
@@ -1,17 +1,10 @@
-cmake_minimum_required(VERSION 3.10.0)
+cmake_minimum_required(VERSION 3.24.0)
 
-set(KTX_FEATURE_TESTS OFF)
-set(KTX_FEATURE_VULKAN OFF)
-set(KTX_FEATURE_GL_UPLOAD OFF)
-set(KTX_FEATURE_TOOLS OFF)
-set(KTX_FEATURE_DOC OFF)
-set(KTX_FEATURE_STATIC_LIBRARY ON)
-
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third_party/glm)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third_party/tinyobjloader)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third_party/KTX-Software)
+project(gfx)
 
 option(GFX_BUILD_EXAMPLES "Build gfx examples" ON)
+option(GFX_ENABLE_GUI "Build gfx with imgui support" OFF)
+option(GFX_ENABLE_SCENE "Build gfx with scene loading support" OFF)
```

Comment:
The 2 new options. Its also possible to add extra finer grained options such as enable-ktx etc. at a later point

Reply:
Issue here I guess is that the `gfx` examples won't build and run out of the box with such a change.
So I think it'd be best to have them on by default.

Reply:
I did update the examples cmake as some will still build without imgui/scene. So when disabled only a subset of the examples will build

Reply:
OK just ran a test to double check. The issue I was trying to avoid by having the options disabled by default is that when they are enabled as default then cmake will start pulling in all the 3rd party dependencies before the user has an option to disable them. So it downloads an additional 1.18GB of dependencies automatically and even if you then disable the option first chance you get its to late as the download has finished at this point. However if the user wants to download dependencies then the first configuration only takes a couple of seconds at which point they can enable the options and then choose to pull in the dependencies without issue and have all examples working
So having the no-dependency option be usable without pulling in all the dependencies automatically on first run anyway, I was thinking is useful for creating minimal test cases etc. for sending in bug examples or for the NN work (such as the one I created last week and used this branch in)

Reply:
Sorry should have been clear, with cmake GUI (emphasis on the GUI) as it does split passes whereas the cmd line does a single pass. So if you run command line and pass to disable GUI/scene from the start then it should work as we would expect. But in the GUI it picks up the default values when you fist load the project and will download all the dependencies before you get the option in the UI to tell it not to. So it depends on what use cases we wanted to support

Reply:
I was thinking of the case of mininn where say they'd already have the cmake file set up correctly at the top level, then a "new developer" who'd check out the project wouldn't get to download these dependencies even if they're enabled by default in gfx, correct?

Reply:
not sure I understand.
Aslong as the cmake options GFX_ENABLE_GUI/GFX_ENABLE_SCENE  are enabled at any point then the dependencies will be downloaded. So if MiniNN pulls in gfx as a submodule with those options disabled in the parent project (providing they are set before the gfx sub project was added) then they would override any defaults in gfx and the dependencies wont be downloaded within that submodule. If a new developer checks out mininn then they would get the same thing as its being overwritten by the minnn parent project. So to get the deps to download the option must be enabled in the parent minn project (this can be done by just ticking the box in cmake GUI - this is why its way easier to enable an option later than it is to disable it as disabling requires clearing caches etc). 
Are you thinking of a new developer who arent familiar with gfx and are learning how to use it so you want them to see the full examples suite? Well either they can just create a standalone gfx clone to play with or do as I mentioned earlier and toggle the option on in the cmake GUI as this wont effect any other users of the project as its a local only change not visible to git